services/fedora/fedora.tester.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+const {
+  isVPlusDottedVersionNClausesWithOptionalSuffixAndEpoch,
+} = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Fedora package (default branch, valid)')
+  .get('/rpm.json')
+  .expectBadge({
+    label: 'fedora',
+    message: isVPlusDottedVersionNClausesWithOptionalSuffixAndEpoch,
+  })
+
+t.create('Fedora package (valid, mocked response)')
```

Comment:
Is this mocked test necessary? It seems like it's exactly the same as the above live test just with a mock response

Reply:
I initially copied the homebrew service tester as basis of this and it had a similar live/mocked same test pair in it. It's also now in the [Debian] and [ArchLinux] testers.

Maybe it in some cases is a good thing in the sense that if the API ever changes, we have some kind of literal example what the expectation was. For [Fedora] that's very well already covered in the schema though, but for example in [Debian] the schema consists of more or less arbitrary patterns, and a written out example helps some.

No strong opinions here, but I think I'd just leave it in for above reason and consistency.

Reply:
You're right that this same thing is done in the tester for `homebrew` but it's still uncommon for us to have have an identical live/mocked test pair. I think the [similar mocked test for Debian](https://github.com/badges/shields/blob/master/services/debian/debian.tester.js#L22) is a bit different (it specifies a non-default value for the distribution), but agree that [ArchLinux](https://github.com/badges/shields/blob/master/services/archlinux/archlinux.tester.js#L15) has an identical live/mock test pair

I'm still inclined to say we should remove this, but @paulmelnikow @chris48s since you looked at some of those other PRs, what's your take?

Reply:
> Maybe it in some cases is a good thing in the sense that if the API ever changes, we have some kind of literal example what the expectation was. For [Fedora] that's very well already covered in the schema though, but for example in [Debian] the schema consists of more or less arbitrary patterns, and a written out example helps some.

Yea, indeed, it's helpful to document what the expectation was. And yea, this is a big part of what the schema is for. I'm not totally sure what you mean by arbitrary patterns. Do you mean the form of the version number?

How about removing the test, and adding a comment alongside the schema, like `// e.g. '4.14.2.1'`?

Reply:
The arbitrary `pattern`s are in [Debian], not here.

I don't think the `version: Joi.string().required()` line here deserves that kind of a comment, but sure, if you think the mock test is more harmful than good, let me know and I'll get rid of it.

Reply:
> The arbitrary `pattern`s are in [Debian], not here.

Ah, gotcha.

Yea, I think the test can be removed.

Reply:
Done.